.github/workflows/conventional-commit-check.yml

```diff
@@ -0,0 +1,37 @@
+name: Semantic Commit Check
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
+
+permissions:
+  contents: read
+
+jobs:
+  semantic-pr:
+    name: Validate PR Title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@2d952a1bf90a6a7ab8f0293dc86f5fdf9acb1915 # v5.5.3
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            feat
+            fix
+            docs
+            style
+            refactor
+            perf
+            test
+            build
+            ci
+            chore
+            revert
+          subjectPattern: ^[A-Z].*[^.]$
+          subjectPatternError: |
+            The PR title subject must start with a capital letter and not end with a period.
```

Comment:
Just to confirm, will this allow `!` to be used along with the types?

Reply:
Yes